File: level2/python/가장 큰 수 42746.py
def solution(num): 
    num = list(map(str, num))
    # 문자열을 3개 나열한 것들('333', '303030', '343434', '555', '999')을 내림차 순 정렬
    # 3개 나열하는 이유 : numbers의 원소는 0 이상 1,000 이하
    num.sort(key = lambda x : x*3, reverse = True) 
    return str(int(''.join(num)))

print(solution([3, 30, 34, 5, 9])) # 테스트 코드

File: level2/javascript/가장 큰 수 42746.js
function solution1(numbers) {  // 방법 1
    let answer = numbers.map((x)=>x+"").sort((a,b)=>a.repeat(3)>b.repeat(3)?-1:1).join("");
    return answer[0] == '0' ? '0' : answer;
}

function solution2(numbers) {  // 방법 2
    let answer = numbers.map((x)=>x + "").sort((a,b)=>(b+a)-(a+b)).join(""); // ex '3','30' => ('303')-('330')
    return answer[0] == '0' ? '0' : answer;
}

// 테스트 코드
console.log(solution2([6,10,2])); // numbers 배열이 0으로만 구성되어 있을 경우 '0'만 출력